hasktorch/src/Torch/TensorFactories.hs

```diff
@@ -54,6 +54,18 @@ linspace start end steps opts = unsafePerformIO $ (cast4 LibTorch.linspace_sslo)
 logspace :: (Scalar a, Scalar b) => a -> b -> Int -> Double -> TensorOptions -> Tensor
 logspace start end steps base opts = unsafePerformIO $ (cast5 LibTorch.logspace_ssldo) start end steps base opts
 
+empty :: [Int] -> TensorOptions -> Tensor
+empty = mkFactoryUnsafe LibTorch.empty_lo
+
+eye :: Int -> TensorOptions -> Tensor
+eye dim opts = unsafePerformIO $ (cast2 LibTorch.eye_lo) dim opts
+
+eyerc :: Int -> Int -> TensorOptions -> Tensor
```

Comment:
`rc` for rectangular?

Reply:
there's probably a better name than eyerc, in the two parameter version you specify both rows and columns (hence rc), whereas in the 1 parameter version it just assumes a square matrix. Open to better ideas.